### ISSUE TITLE
[Feature Request] Global Options/Configs

### ISSUE THREAD

**Issue by spigelli** [PrismLauncher/PrismLauncher#1077]
It would be nice if there were an option for users to specify a global options.txt so that keybinds can be shared between instances. I play _ESDF_ rather than _WASD_ and in general have a bunch of strange binds and setting that I set for every instance and it becomes cumbersome to bootstrap each instance by going through each setting or copying over an options.txt from another instance.

Related potential features:
- Per instance-group `options.txt`
- Global/instance-group mod configs
- `.minecraft/config/<mod>/<mod>.<toml|json>` merging

**Comment by O4bit** [comment]
I agree 
(i also had the idea to use packwiz but for most people it would be very complicated)

**Comment by Stein-N** [comment]
Or like the Curseforge app does it with the most basic settings every user seems to change as soon you start an instance for the first time:

![Image](https://github.com/user-attachments/assets/bf88a60d-effc-4b3a-803e-76a6bd340f95)

**Comment by Trial97** [comment]
The option part is a duplicate of: https://github.com/PrismLauncher/PrismLauncher/issues/380
The other regarding mods config may be considered once #38 is resolved(by making a symlink.
With that said please move the suggestion regarding the options file to the other issue

**Comment by smokcat** [comment]
i have a working (but hacky & linux only) prototype of this:

this file needs to be executed as a pre-launch-command by prismlauncher:
[~/.local/share/PrismLauncher/instances/default_options.sh](https://github.com/user-attachments/files/25461083/default_options.sh)
```bash
#!/bin/bash

if [ -f "$INST_MC_DIR/options.txt" ]
then
    echo "it exists, nothing to do here"
else
    cp "$HOME/.local/share/PrismLauncher/instances/options.txt" "$INST_MC_DIR/options.txt"
fi
```

and this is the config it will insert into every profile that doesnt contain a `options.txt` at launch (newly created profiles & ones where the file has been deleted somehow)
[~/.local/share/PrismLauncher/instances/options.txt](https://github.com/user-attachments/files/25461082/options.txt)
```
enableVsync:true
fullscreen:true
guiScale:3
maxFps:60
renderDistance:32
soundCategory_music:0.0
```
note1: the example config contains my personal preferences, but it can contain any valid entry, as minecraft itself handles the parsing and insertion of unspecified values automatically.

note2: the above script should absolutely work within windows, if someone re-writes it in cmd or powershell, not that having it as an external script is a good solution (just a working one for now, with many edgecased i would guess)

**Comment by puditcool** [comment]
Make so that you are able to toggle in the launcher so that every time you create a new instances, the settings (controls, keybinds, resource packs, shaders, screenshots, FOV, etc) Stay the same. something like Shared Resources mod by enjarai.

When i create multiple instances i find it annoying when i create a new instance and i have to change my keybinds, mouse sensitivity, FOV, etc. every time i create a new instances.

**Comment by UsefullSpoon** [comment]
and perhaps a way to also transfer your servers over

**Comment by LeoMerlino** [comment]
Any work on this?
Every time I create a new instance I have to manually copy over the options.txt file which is quite cumbersome.
Having some sort of global config that gets applied automatically to new instances would be really nice.

**Comment by amqndin** [comment]
from the outside at least, this seems like a pretty simple feature to implement, that everyone would benefit from. but there has been no progress in 4 years?? this is the third issue with the exact same feature request, clearly, this has been in high demand, not complicated, all benefit and no downside. why would this get ignored for the last 4 years straight?

**Comment by amqndin** [comment]
nope, i was incorrect, that was just the open issues, but at least 5 more issues closed as duplicates. so 8 issues requesting the same feature in the last 4 years
